puppet/modules/jenkins_job_builder/files/centos.org/ansible/provision.yml

```diff
@@ -3,12 +3,25 @@
   vars:
     api_key: "{{ lookup('file', '~/duffy.key') }}"
   tasks:
-    - name: 'Get list of nodes'
+    - name: 'Get a new node'
       cico:
         action: get
         api_key: "{{ api_key }}"
       register: cico_data
 
+    - name: 'Retry geting list of nodes'
```

Comment:
Is the idea that the host was actually provisioned, but the return data did not provide useful information? Or that the host was never actually provisioned?

Reply:
I've seen jobs where the `cico` module returns success, but the hosts list is empty. I don't know if that's a race condition, and the host is provisioned later, or if the API wrongly returned success (which is more likely, but I didn't have the chance to poke at the API enough yet).

My idea was to at least retry the listing of hosts and fail loudly if that also fails.

The fail is actually more important here, as it makes the Jenkins job fail nicely, instead of a weird "no hosts matched" warning from ansible.

Reply:
OK, sounds like we just need to keep an eye on things after this.